### PR TITLE
Custom markdown reflink tokenizer for broken link detection

### PR DESCRIPTION
**Pull request by csauve**
This PR upgrades our `marked` dependency which allows us to override the tokenizer used when parsing reflinks, allowing for some interesting possibilities. As a reminder, reflinks are this:

```md
This is [a reflink][ref]. Please see the [tags][] page.

[ref]: www.example.com
[tags]: /h1/tags
```

The previous implementation generated a list of every possible page and heading combination on the site and appended them to the tail of every piece of markdown rendered, which was a bit of a hack:

```md
[h1]: /h1
[h1#editions-and-versions]: /h1#editions-and-versions
[h1#xbox]: /h1#xbox
[scenario]: /h1/tags/scenario
[scenario#child-scenarios]: /h1/tags/scenario#child-scenarios
...etc
```

The new code constructs a custom reflink tokenizer in the context of the `metaIndex` API, which allows it to use `metaIndex.resolvePage(slug, heading)` to look up pages and headings dynamically. The user will still be able to define reflinks in the body of their markdown and override implicit reflink lookup. The benefit here is that `metaIndex.resolvePage()` throws errors when the page and heading cannot be found. This results in broken link detection, a few being fixed in this PR.

Also in this PR I organized the markdown rendering code into more files for clarity.

### CODE REVIEW

File: src/content.js [hunk 1]
```diff
@@ -123,14 +123,6 @@ async function buildMetaIndex(contentDir, invaderDefsDir, baseUrl, packageVersio
   const invaderStructDefs = await getInvaderStructDefs(invaderDefsDir);
   const data = buildData(invaderStructDefs);
 
-  const mdFooter = pages
-    .filter(page => page._pathParts.length > 0)
-    .flatMap(page => [
-      `[${page._slug}]: ${page._path}`,
-      ...page._headers.map(header => `[${page._slug}#${header.id}]: ${page._path}#${header.id}`)
-    ])
-    .join("\n");
```
Comment (csauve) [inline src/content.js:132]: This was the giant all-pages all-headers reflink section we appended before.

File: src/content/h1/readme.md [hunk 2]
```diff
@@ -87,7 +87,7 @@ In 2011, Halo: Combat Evolved Anniversary was released for Xbox 360. Often calle
 ## MCC (PC and Xbox One)
 Halo: The Master Chief collection (MCC) is actively maintained by [343 Industries][343i] for both PC and Xbox One. It brings the Halo series under a single [Game as a Service][gaas], including unified matchmaking and progression experiences. The PC port uses [Unreal Engine][unreal] as a menu and input layer over the respective engines of each included Halo game.
 
-It is possible to compile custom maps for MCC PC using Invader and some [experimental tooling][opack].
+It is possible to compile custom maps for MCC PC using older versions of Invader.
```
Comment (csauve) [inline src/content/h1/readme.md:90]: broken link -- we don't have an opack page

File: src/data/h1/tags/scenario.yml [hunk 3]
```diff
@@ -700,7 +700,7 @@ comments:
       md: |
         Axis-aligned volumes which can be used in scripts to test for the
         presence an object: `(volume_test_object "volume_name" <object>)`.
-        An example use case is [kill volumes][kill-volume].
+        An example use case is [kill volumes][kill-volumes].
```
Comment (csauve) [inline src/data/h1/tags/scenario.yml:703]: broken link -- typo

File: src/content/h1/readme.md (hunk 2 above)
Comment (gbMichelle) [reply src/content/h1/readme.md:90]: Shouldn't opack still be mentioned?

Comment (csauve) [reply src/content/h1/readme.md:90]: No, the repo got deleted by its author and its obsolete with the stuff Zatarita is working on. Suppose I could mention that tool but don’t know enough about it to do it justice yet on the wiki

Comment (gbMichelle) [reply src/content/h1/readme.md:90]: Oh yeah sorry then. Forgot that was another project

**Pull request merged**